tasties_app/tests/test_authentication.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+from django.contrib.auth.models import User
+from pytest_django.asserts import assertTemplateUsed
+from authConstants import (
+    VALID_EMAIL, VALID_PASSWORD, VALID_USER, INVALID_EMAIL, INVALID_PASSWORD_MISMATCH, LOGIN_PATH, REGISTER_PATH,
+    MISMATCH_ERROR
+)
+
+
+@pytest.mark.django_db
+class TestViews:
+    @pytest.fixture
+    def signed_up_credentials(self):
+        user = User.objects.create_user(username=VALID_USER, email=VALID_EMAIL, password=VALID_PASSWORD)
+        user.save()
+
+    def test_login_loaded(self, client):
+        response = client.get('/login/')
+        assert response.status_code == 200
+        assertTemplateUsed(response, LOGIN_PATH)
+
+    def test_register_loaded(self, client):
+        response = client.get('/register/')
+        assert response.status_code == 200
+        assertTemplateUsed(response, REGISTER_PATH)
+
+    def test_valid_registration(self, client):
+        response = client.post('/register/', data={'username': 'telhaiuser', 'email': 'user@telhai.ac.il',
+                                                   'password1': 'tastiesPassword', 'password2': 'tastiesPassword'})
+        assert response.status_code == 302
+        assert response.url == '/login/'
+
+    def test_invalid_email_registration(self, client):
+        response = client.post('/register/', data={'username': VALID_USER, 'email': INVALID_EMAIL,
+                                                   'password1': VALID_PASSWORD, 'password2': VALID_PASSWORD})
+        assert response.status_code == 200
+        assert b"Register Account" in response.content
+
+    def test_invalid_password_registration(self, client):
+        response = client.post('/register/', data={'username': VALID_USER, 'email': VALID_EMAIL,
+                                                   'password1': VALID_PASSWORD, 'password2': INVALID_PASSWORD_MISMATCH})
+        assert response.status_code == 200
+        assert response.context['form'].error_messages == MISMATCH_ERROR
+
+    def test_valid_login(self, client, signed_up_credentials):
+        response = client.post('/login/', data={'username': VALID_USER, 'password': VALID_PASSWORD})
+        assert response.status_code == 302
+        assert response.url == '/'
+
+    def test_invalid_login(self, client, signed_up_credentials):
+        response = client.post('/login/', data={'username': VALID_USER, 'password': INVALID_PASSWORD_MISMATCH})
+        assert response.status_code == 200
```

Comment:
I tried to show here that I'm staying on the same page instead of being redirected to another page (302).
So if I assert `response.status_code != 302` will it be better?

Reply:
There is no redirect? how the user should know that there is an error?

Reply:
No, you can see in the `register` method in `views.py` that if the form is invalid it will add an error message to `messages`, the page will be rendered, and then the error will bump. This is why the status code will be 200.

Reply:
ok, thanks

Reply:
So should I use `assert response.status_code != 302` To make it more readable?

Reply:
I think it's better to keep it with status code 200 and make sure it's the correct template